File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        count = defaultdict(int)

        if len(s)!=len(t):
            return False
        
        for i in range(len(s)):
            count[s[i]] += 1
            count[t[i]] -= 1
        
        for k in count.keys():
            if count[k]!=0:
                return False
        
        return True

File: 0242-valid-anagram/0242-valid-anagram.c
bool isAnagram(char* s, char* t) {
    if (strlen(s) != strlen(t))
        return false;

    int count[26] = {0};

    for (int i=0; i<strlen(s); ++i) {
        count[s[i]-'a']++;
        count[t[i]-'a']--;
    }

    for (int i=0; i<26; ++i) {
        if (count[i] != 0)
            return false;
    }

    return true;
}